x-pack/plugins/ml/public/application/timeseriesexplorer/components/forecasting_modal/run_controls.js

```diff
@@ -134,34 +137,60 @@ export function RunControls({
-              label={
-                <FormattedMessage
-                  id="xpack.ml.timeSeriesExplorer.runControls.durationLabel"
-                  defaultMessage="Duration"
-                />
-              }
-              fullWidth
-              isInvalid={!isNewForecastDurationValid}
-              error={newForecastDurationErrors}
-              helpText={
-                <FormattedMessage
-                  id="xpack.ml.timeSeriesExplorer.runControls.forecastMaximumLengthHelpText"
-                  defaultMessage="Length of forecast, up to a maximum of {maximumForecastDurationDays} days.
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup direction="column">
+              <EuiFlexItem grow={false}>
+                <EuiFormRow
+                  label={
+                    <FormattedMessage
+                      id="xpack.ml.timeSeriesExplorer.runControls.durationLabel"
+                      defaultMessage="Duration"
+                    />
+                  }
+                  fullWidth
+                  isInvalid={!isNewForecastDurationValid}
+                  error={newForecastDurationErrors}
+                  helpText={
+                    <FormattedMessage
+                      id="xpack.ml.timeSeriesExplorer.runControls.forecastMaximumLengthHelpText"
+                      defaultMessage="Length of forecast, up to a maximum of {maximumForecastDurationDays} days.
                   Use s for seconds, m for minutes, h for hours, d for days, w for weeks."
-                  values={{ maximumForecastDurationDays: FORECAST_DURATION_MAX_DAYS }}
-                />
-              }
-            >
-              {disabledState.isDisabledToolTipText === undefined ? (
-                durationInput
-              ) : (
-                <EuiToolTip position="right" content={disabledState.isDisabledToolTipText}>
-                  {durationInput}
-                </EuiToolTip>
-              )}
-            </EuiFormRow>
+                      values={{ maximumForecastDurationDays: FORECAST_DURATION_MAX_DAYS }}
+                    />
+                  }
+                >
+                  {disabledState.isDisabledToolTipText === undefined ? (
+                    durationInput
+                  ) : (
+                    <EuiToolTip position="right" content={disabledState.isDisabledToolTipText}>
+                      {durationInput}
+                    </EuiToolTip>
+                  )}
+                </EuiFormRow>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiFormRow
+                  helpText={i18n.translate(
+                    'xpack.ml.timeSeriesExplorer.runControls.neverExpireHelpText',
+                    {
+                      defaultMessage:
+                        'When not selected, created forecasts will expire after 14 days.',
+                    }
+                  )}
+                >
+                  <EuiSwitch
```

Comment:
You'll need to disable this switch if the user doesn't have permission to run forecasts? Or I wonder if just hide the entire 'Run a new forecast section'?

Reply:
The switch is disabled if the 'Run' is disabled. That is existing behavior. If we want to hide it completely that would require an additional change in the parent component. 🤔 I'm not sure we should change it since it has been available to be viewed until now.

Reply:
All good. Just wanted to make sure that the 'Never expire' switch was enabled as well as the 'Run' button.